sdk/communication/communication-common/test/public/communicationKeyCredentialPolicy.spec.ts

```diff
@@ -58,6 +59,128 @@ describe("CommunicationKeyCredentialPolicy", function () {
+        if (responses.length) {
+          const response = responses.shift()!;
+          response.request = req;
+          return response;
+        }
+        throw new Error("No responses found");
+      },
+    };
+
+    const pipeline = createEmptyPipeline();
+    pipeline.addPolicy(communicationKeyCredentialPolicy);
+    const pipelineResponse = await pipeline.sendRequest(testHttpsClient, pipelineRequest);
+
+    const authHeader = pipelineResponse.request.headers.get("Authorization");
+    const dateHeader = pipelineResponse.request.headers.get("x-ms-date");
+    const hashHeader = pipelineResponse.request.headers.get("x-ms-content-sha256");
+    const hostHeader = pipelineResponse.request.headers.get("Host");
+
+    const authHeaderExpected = await generateAuthHeader(
+      pipelineRequest.method.toLocaleUpperCase(),
+      dateHeader || "",
+      credential.key,
+      "/testPath?testQuery=",
+      "example.com"
+    );
+
+    assert.equal(authHeader, authHeaderExpected);
+    assert.isNotEmpty(dateHeader);
+    assert.isNotEmpty(hashHeader);
+    if (isNode) {
+      assert.isNotEmpty(hostHeader);
+    }
+  });
+});
+
+describe("CommunicationKeyCredentialPolicy", function () {
+  it("signs the request correctly with path and no query", async function () {
+    const credential = new MockKeyCredential("pw==");
+    const communicationKeyCredentialPolicy =
+      createCommunicationAccessKeyCredentialPolicy(credential);
+
+    const pipelineRequest = createPipelineRequest({ url: "https://example.com/testPath" });
+
+    const responses: PipelineResponse[] = [
+      {
+        headers: createHttpHeaders(),
+        request: pipelineRequest,
+        status: 200,
+      },
+    ];
+
+    const testHttpsClient: HttpClient = {
+      sendRequest: async (req) => {
+        if (responses.length) {
+          const response = responses.shift()!;
+          response.request = req;
+          return response;
+        }
+        throw new Error("No responses found");
+      },
+    };
+
+    const pipeline = createEmptyPipeline();
+    pipeline.addPolicy(communicationKeyCredentialPolicy);
+    const pipelineResponse = await pipeline.sendRequest(testHttpsClient, pipelineRequest);
+
+    const authHeader = pipelineResponse.request.headers.get("Authorization");
+    const dateHeader = pipelineResponse.request.headers.get("x-ms-date");
+    const hashHeader = pipelineResponse.request.headers.get("x-ms-content-sha256");
+    const hostHeader = pipelineResponse.request.headers.get("Host");
+
+    const authHeaderExpected = await generateAuthHeader(
+      pipelineRequest.method.toLocaleUpperCase(),
+      dateHeader || "",
+      credential.key,
+      "/testPath",
+      "example.com"
+    );
+
+    assert.equal(authHeader, authHeaderExpected);
+    assert.isNotEmpty(dateHeader);
+    assert.isNotEmpty(hashHeader);
+    if (isNode) {
+      assert.isNotEmpty(hostHeader);
+    }
+  });
+});
+
+async function generateAuthHeader(
```

Comment:
Instead of this function, I'd create auth header fixtures for different set of params - `/testPath` and `/testPath?testQuery=`. This might require a little bit of refactoring in the `createCommunicationAccessKeyCredentialPolicy` but I think it's worth it. Otherwise, we're testing one function against another function.

Reply:
Please let me know if the updated unit tests are okay